File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/domain/entity/Ingredient.java
package com.tarbadev.witchcraft.domain.entity;

import lombok.*;

@Data
@Builder(toBuilder = true)
@AllArgsConstructor
@NoArgsConstructor
public class Ingredient {
  private Integer id;
  private String name;
  private Double quantity;
  private String unit;

  public Ingredient addQuantity(Double quantity) {
    return this.toBuilder()
        .quantity(this.quantity + quantity)
        .build();
  }
}

File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/recipes/domain/entity/Ingredient.kt
package com.tarbadev.witchcraft.recipes.domain.entity

data class Ingredient (
    val id: Int = 0,
    val name: String = "",
    val quantity: Double = 0.0,
    val unit: String = ""
) {

    fun addQuantity(quantity: Double): Ingredient {
        return this.copy(quantity = this.quantity + quantity)
    }
}
